src/node/biquad_filter.rs

```diff
@@ -0,0 +1,1180 @@
+            .send(CoeffsReq(sender))
+            .expect("Sending CoeffsReq failed");
+
+        loop {
+            match receiver.try_recv() {
+                Err(crossbeam_channel::TryRecvError::Disconnected) => {
+                    println!("Receiver Error: disconnected type");
+                    continue;
+                }
+                Err(crossbeam_channel::TryRecvError::Empty) => {
+                    println!("Receiver Error: empty type");
+                    continue;
+                }
+                Ok([b0, b1, b2, a1, a2]) => {
+                    println!("received...");
+                    for (i, &f) in frequency_hz.iter().enumerate() {
+                        let f = f64::from(f);
+                        let sample_rate = f64::from(self.sample_rate);
+                        let num = b0
+                            + Complex::from_polar(b1, -1.0 * 2.0 * PI * f / sample_rate)
+                            + Complex::from_polar(b2, -2.0 * 2.0 * PI * f / sample_rate);
+                        let denom = 1.0
+                            + Complex::from_polar(a1, -1.0 * 2.0 * PI * f / sample_rate)
+                            + Complex::from_polar(a2, -2.0 * 2.0 * PI * f / sample_rate);
+                        let h_f = num / denom;
+
+                        // Possible truncation is fine. f32 precision should be sufficients
+                        // And it is required by the specs
+                        mag_response[i] = h_f.norm() as f32;
+                        phase_response[i] = h_f.arg() as f32;
+                    }
+                    break;
+                }
+            }
+        }
+    }
+
+    #[inline]
+    fn validate_inputs(
+        &self,
+        frequency_hz: &mut [f32],
+        mag_response: &mut [f32],
+        phase_response: &mut [f32],
+    ) {
+        assert_eq!(
+            frequency_hz.len(),
+            mag_response.len(),
+            " InvalidAccessError: All paramaters should be the same length"
+        );
+        assert_eq!(
+            mag_response.len(),
+            phase_response.len(),
+            " InvalidAccessError: All paramaters should be the same length"
+        );
+
+        // Ensures that given frequencies are in the correct range
+        let min = 0.;
+        let max = self.sample_rate / 2.;
+        for f in frequency_hz.iter_mut() {
+            *f = f.clamp(min, max);
+        }
+    }
+
+    /// Mock of `get_frequency_response`
+    /// This function is the same as `get_frequency_response` except it never send the `CoeffsReq`.
+    /// In tests, we use `OfflineAudioContext` and in this context the `CoeffsReq` is not sendable.
+    ///
+    /// # Arguments
+    ///
+    /// * `frequency_hz` - frequencies for which frequency response of the filter should be calculated
+    /// * `mag_response` - magnitude of the frequency response of the filter
+    /// * `phase_response` - phase of the frequency response of the filter
+    #[cfg(test)]
```

Comment:
`#[cfg(not(test))]` is not well supported by rust analyzer. This is why i made a mock function to use `#[cfg(test)]` instead